### PR TITLE
Blokh/feat/task level definitions ( READY FOR REVIEW )

### PR DESCRIPTION
**Pull request by Blokh**
- [x] added defintion level rules
- [x] added deliver event logic via workflow service controllers on - update workflow & update decision
- [x] update FED to deliver event
- [x] updated seed
- [x] update common
- [x] Release new common package

### CODE REVIEW

File: apps/backoffice-v2/src/pages/Entity/components/CallToAction/hooks/useCallToActionLogic/useCallToActionLogic.tsx [hunk 1]
```diff
@@ -8,20 +8,27 @@ import { useCallback, useEffect, useState } from 'react';
 import { useApproveTaskByIdMutation } from '../../../../../../domains/entities/hooks/mutations/useApproveTaskByIdMutation/useApproveTaskByIdMutation';
 import { useRejectTaskByIdMutation } from '../../../../../../domains/entities/hooks/mutations/useRejectTaskByIdMutation/useRejectTaskByIdMutation';
 import { useRevisionTaskByIdMutation } from '../../../../../../domains/entities/hooks/mutations/useRevisionTaskByIdMutation/useRevisionTaskByIdMutation';
+import { TWorkflowById } from '../../../../../../domains/workflows/fetchers';
 
+const getDeliverableEvent = (workflow: TWorkflowById) => {
+  if (workflow?.workflowDefinition?.config?.workflowLevelResolution) {
+    return 'DOCUMENT_REVIEWED';
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/pages/Entity/components/CallToAction/hooks/useCallToActionLogic/useCallToActionLogic.tsx]: Let's use a shared `as const` object here.

File: apps/backoffice-v2/src/pages/Entity/components/CallToAction/hooks/useCallToActionLogic/useCallToActionLogic.tsx [hunk 2]
```diff
@@ -8,20 +8,27 @@ import { useCallback, useEffect, useState } from 'react';
 import { useApproveTaskByIdMutation } from '../../../../../../domains/entities/hooks/mutations/useApproveTaskByIdMutation/useApproveTaskByIdMutation';
 import { useRejectTaskByIdMutation } from '../../../../../../domains/entities/hooks/mutations/useRejectTaskByIdMutation/useRejectTaskByIdMutation';
 import { useRevisionTaskByIdMutation } from '../../../../../../domains/entities/hooks/mutations/useRevisionTaskByIdMutation/useRevisionTaskByIdMutation';
+import { TWorkflowById } from '../../../../../../domains/workflows/fetchers';
 
+const getDeliverableEvent = (workflow: TWorkflowById) => {
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/pages/Entity/components/CallToAction/hooks/useCallToActionLogic/useCallToActionLogic.tsx]: Let's move this function to a different file.

File: apps/backoffice-v2/src/pages/Entity/hooks/useTasks/useTasks.tsx [hunk 3]
```diff
@@ -28,6 +28,12 @@ const motionProps: ComponentProps<typeof MotionBadge> = {
   animate: { y: 0, opacity: 1, transition: { duration: 0.2 } },
 };
 
+function getDeliverableEvent(workflow: TWorkflowById): string | undefined {
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/pages/Entity/hooks/useTasks/useTasks.tsx]: Let's either rename the identical functions, share the same function, or add a comment to explain the duplicated logic.

File: apps/backoffice-v2/src/domains/entities/hooks/mutations/useRemoveDecisionTaskByIdMutation/useRemoveDecisionTaskByIdMutation.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+            documents: oldWorkflow?.context?.documents?.map(document => {
+              if (document?.id !== documentId) {
+                return document;
+              }
+
+              return {
+                ...document,
+                decision: {
+                  ...document?.decision,
+                  status: 'revision',
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/domains/entities/hooks/mutations/useRemoveDecisionTaskByIdMutation/useRemoveDecisionTaskByIdMutation.tsx]: Optimistic updates should reflect the expected response.
```suggestion
                  status: null,
```

File: services/workflows-service/src/workflow/workflow.service.ts [hunk 5]
```diff
@@ -1589,11 +1559,12 @@ export class WorkflowService {
     workflowDefinition: WorkflowDefinition,
     context: DefaultContextSchema,
   ) {
+    console.log('running validate');
     if (!Object.keys(workflowDefinition?.contextSchema ?? {}).length) return;
-
+    console.log('passed schema');
```
Comment (Omri-Levy) [inline services/workflows-service/src/workflow/workflow.service.ts]: Make sure to remove added instances of `console.log`.

File: services/workflows-service/src/workflow/workflow.service.ts [hunk 6]
```diff
@@ -1593,7 +1563,7 @@ export class WorkflowService {
 
     const validate = ajv.compile(workflowDefinition?.contextSchema?.schema); // TODO: fix type
     const isValid = validate(context);
-
+    console.log(JSON.stringify(context));
```
Comment (alonp99) [inline services/workflows-service/src/workflow/workflow.service.ts]: Remove

File: services/workflows-service/src/workflow/workflow.controller.external.ts [hunk 7]
```diff
@@ -221,7 +221,7 @@ export class WorkflowControllerExternal {
     @ProjectIds() projectIds: TProjectIds,
     @CurrentProject() currentProjectId: TProjectId,
   ): Promise<void> {
-    return await this.service.event(
+    await this.service.event(
```
Comment (alonp99) [inline services/workflows-service/src/workflow/workflow.controller.external.ts]: Curious about this change

Comment (Blokh) [reply services/workflows-service/src/workflow/workflow.controller.external.ts]: send event now return updated runtime after the event was sent with the new snapshot for updated machine state ( as it calls updateWorkflowRuntime and returns the output of it

**Review by alonp99 — COMMENTED** [review]
App

**Pull request merged**